### PR TITLE
Fixed teleporting when no hit is found.

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Updated valid calculation up direction
Updated distance calculation used when rendering line.

Changes
---
- Fixes: #2591

### CODE REVIEW

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs [hunk 1]
```diff
@@ -195,7 +196,7 @@ public override void OnPostRaycast()
                         {
                             // If it's NOT a hotspot, check if the hit normal is too steep 
                             // (Hotspots override dot requirements)
-                            TeleportSurfaceResult = Vector3.Dot(Result.StartPoint, Vector3.up) < upDirectionThreshold
+                            TeleportSurfaceResult = Vector3.Dot(Result.Details.LastRaycastHit.normal, Vector3.up) > upDirectionThreshold
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: can devs override this threshold? for example if they want to make an experience where the user climbs steep inclines?

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs [hunk 2]
```diff
@@ -350,7 +352,12 @@ public override void OnPositionInputChanged(InputEventData<Vector2> eventData)
                 {
                     canTeleport = false;
                     teleportEnabled = false;
-                    TeleportSystem.RaiseTeleportStarted(this, TeleportHotSpot);
+
+                    if (TeleportSurfaceResult == TeleportSurfaceResult.Valid ||
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: I see distance being calculated but it is not being considered here... is that intentional? is distance a factor in whether or not a teleport should be allowed?

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 1 above)
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: Yes by setting the value in the inspector.

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 2 above)
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: At this point we've already calculated a valid point. So I'm not sure what you mean here.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: was just wondering if distance was involved in determining a valid point.

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs [hunk 3]
```diff
@@ -224,7 +225,7 @@ public override void OnPostRaycast()
                             }
 
                             // Only add the distance between the start point and the hit
-                            clearWorldLength += Vector3.Distance(Result.StartPoint, Result.StartPoint);
+                            clearWorldLength += Vector3.Distance(Result.StartPoint, Result.Details.Point);
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: @davidkline-ms this distance calculation is only for properly clamping the line renderer to properly render to the end of the parabola. It is not used in calculating the validity of the teleport result and is outside the scope of this PR.

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs (hunk 2 above)
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/TeleportPointer.cs]: It is not

**Comment by StephenHodgson** [comment]
Bump

**Pull request merged**